app/src/main/java/com/google/samples/apps/nowinandroid/ui/AdaptiveScaffold.kt

```diff
@@ -0,0 +1,220 @@
+    val drawerContainerColor: Color,
+    val bottomBarContainerColor: Color,
+    val contentContainerColor: Color,
+    val contentColor: Color,
+    val selectedIconColor: Color,
+    val selectedTextColor: Color,
+    val unselectedIconColor: Color,
+    val unselectedTextColor: Color,
+    val selectedContainerColor: Color,
+    val unselectedContainerColor: Color,
+)
+
+object AdaptiveScaffoldNavigationComponentDefaults {
+    @Composable
+    fun colors(
+        railContainerColor: Color = NavigationRailDefaults.ContainerColor,
+        drawerContainerColor: Color = MaterialTheme.colorScheme.surface,
+        bottomBarContainerColor: Color = NavigationBarDefaults.containerColor,
+        contentContainerColor: Color = MaterialTheme.colorScheme.background,
+        contentColor: Color = contentColorFor(contentContainerColor),
+        selectedIconColor: Color = MaterialTheme.colorScheme.onSecondaryContainer,
+        selectedTextColor: Color = MaterialTheme.colorScheme.onSecondaryContainer,
+        unselectedIconColor: Color = MaterialTheme.colorScheme.onSurfaceVariant,
+        unselectedTextColor: Color = MaterialTheme.colorScheme.onSurfaceVariant,
+        selectedContainerColor: Color = MaterialTheme.colorScheme.secondaryContainer,
+        unselectedContainerColor: Color = MaterialTheme.colorScheme.surface,
+    ) = AdaptiveScaffoldNavigationComponentColors(
+        railContainerColor = railContainerColor,
+        drawerContainerColor = drawerContainerColor,
+        bottomBarContainerColor = bottomBarContainerColor,
+        contentContainerColor = contentContainerColor,
+        contentColor = contentColor,
+        selectedIconColor = selectedIconColor,
+        selectedTextColor = selectedTextColor,
+        unselectedIconColor = unselectedIconColor,
+        unselectedTextColor = unselectedTextColor,
+        selectedContainerColor = selectedContainerColor,
+        unselectedContainerColor = unselectedContainerColor,
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun <T> AdaptiveScaffold(
+    modifier: Modifier = Modifier,
+    navigationItems: List<T>,
+    navigationItemTitle: @Composable (item: T, isSelected: Boolean) -> Unit,
+    navigationItemIcon: @Composable (item: T, isSelected: Boolean) -> Unit,
+    isItemSelected: @Composable (item: T) -> Boolean,
+    onNavigationItemClick: (item: T) -> Unit,
+    topBar: @Composable () -> Unit = {},
+    snackbarHost: @Composable () -> Unit = {},
+    colors: AdaptiveScaffoldNavigationComponentColors = AdaptiveScaffoldNavigationComponentDefaults.colors(),
+    contentWindowInsets: WindowInsets = ScaffoldDefaults.contentWindowInsets,
+    content: @Composable (padding: PaddingValues) -> Unit,
+) {
+    val context = LocalContext.current
+    val metrics = WindowMetricsCalculator.getOrCreate().computeCurrentWindowMetrics(context)
+    val widthDp = metrics.bounds.width() / context.resources.displayMetrics.density
+    val movableContent = remember(content) {
```

Comment:
I don't think we want to be using `content` as a key here, since we don't want to wipe away the whole state if the `content` lambda from the caller changes slightly. Although I'm not really sure how `@Composable (padding: PaddingValues) -> Unit` equality works to begin with. If `AppState` changes or something else changes in what we pass to `AdaptiveScaffold`, will that cause `content` to be different?

Reply:
Hey @alexvanyo, using content as a key is a recommendation from our docs:

https://developer.android.com/reference/kotlin/androidx/compose/runtime/package-summary#movableContentOf(kotlin.Function0)